modules/System Admin/logs_view.php

```diff
@@ -0,0 +1,131 @@
+use Gibbon\Domain\DataSet;
+use Gibbon\Tables\DataTable;
+use Gibbon\Services\Format;
+use Gibbon\Domain\System\LogGateway;
+
+require_once __DIR__ . '/moduleFunctions.php';
+
+if (isActionAccessible($guid, $connection2, '/modules/System Admin/logs_view.php') == false) {
+    //Acess denied
+    echo "<div class='error'>";
+    echo __('You do not have access to this action.');
+    echo '</div>';
+} else {
+    //Proceed!
+    $page->breadcrumbs->add(__('View Logs'));
+
+    if (isset($_GET['return'])) {
+        returnProcess($guid, $_GET['return'], null, $returns);
+    }
+
+    $ip = isset($_GET['ip'])? $_GET['ip'] : '';
+    $title = isset($_GET['title'])? $_GET['title'] : '';
+    $gibbonPersonID = isset($_GET['gibbonPersonID'])? $_GET['gibbonPersonID'] : '';
+
+    echo '<h2>';
+    echo __('Filters');
+    echo '</h2>';
+
+    $form = Form::create('filter', $_SESSION[$guid]['absoluteURL'].'/index.php', 'get');
+
+    $form->setFactory(DatabaseFormFactory::create($pdo));
+    $form->setClass('noIntBorder fullWidth');
+    $form->addHiddenValue('q', '/modules/'.$_SESSION[$guid]['module'].'/logs_view.php');
+
+    $data = array();
+    $sql = "SELECT DISTINCT title AS value, title AS name FROM gibbonLog ORDER BY title";
+    $row = $form->addRow();
+        $row->addLabel('title', __('Title'));
+        $row->addSelect('title')->fromQuery($pdo, $sql, $data)->selected($title)->placeholder();
+
+    $row = $form->addRow();
+        $row->addLabel('gibbonPersonID', __('User'));
+        $row->addSelectUsers('gibbonPersonID')->selected($gibbonPersonID)->placeholder();
+
+    $row = $form->addRow();
+        $row->addLabel('ip', __('IP Address'))
+            ->setClass('mediumWidth');
+        $row->addTextField('ip')->setValue($ip);
+
+    $row = $form->addRow();
+        $row->addSearchSubmit($gibbon->session, __('Clear Search'));
+
+    echo $form->getOutput();
+
+    echo '<h2>';
+    echo __('Data');
+    echo '</h2>';
+
+    // QUERY
+    $logGateway = $container->get(LogGateway::class);
+    $criteria = $logGateway->newQueryCriteria(true)
+        ->sortBy('timestamp', 'DESC')
+        ->filterBy('ip', $ip)
+        ->filterBy('title', $title)
+        ->filterBy('gibbonPersonID', $gibbonPersonID)
+        ->fromPOST();
+
+    $logs = $logGateway->queryLogs($criteria);
+
+    $table = DataTable::createPaginated('logView', $criteria);
+
+    // $table->addMetaData('filterOptions', [
+    //     'type:core'       => __('Type').': '.__('Core'),
+    //     'type:additional' => __('Type').': '.__('Additional'),
+    //     'active:Y' => __('Active').': '.__('Yes'),
+    //     'active:N' => __('Active').': '.__('No'),
+    // ]);
+
+    $table->addExpandableColumn('comment')
+        ->format(function($log) {
+            $array = unserialize($log['serialisedArray']);
+
+            $details = '';
+            if (count($array) > 0) {
+                $details = "<table class='smallIntBorder' style='width:100%;'>";
```

Comment:
🤔

Reply:
You don't like my crazy table-within-a-table? I copied it from Browse Library. Open to better ways to do this!